Maskbook/Scene/App/RedPacket/ViewModel/LuckyDropViewModel.swift

```diff
@@ -0,0 +1,728 @@
+                raiseOnDivideByZero: false)
+            result = result.rounding(accordingToBehavior: roundBehavior)
+        } else {
+            result = NSDecimalNumber(string: amountStr)
+        }
+        return result != .notANumber ? result : .zero
+    }
+    
+    var totalQuantityStr: String {
+        let roundBehavior = NSDecimalNumberHandler(roundingMode: .plain, scale: 2, raiseOnExactness: false, raiseOnOverflow: false, raiseOnUnderflow: false, raiseOnDivideByZero: false)
+        let result = totalQuantity.rounding(accordingToBehavior: roundBehavior)
+        guard result != .notANumber else {
+            return "0.00"
+        }
+        let doubleValue = totalQuantity.doubleValue
+        if doubleValue < 0.01, doubleValue > 0 {
+            return "< 0.01"
+        }
+        return String(format: "%.2f", result.doubleValue)
+    }
+    
+    var totalQuantityColor: Color {
+        totalQuantity == .zero ?
+            Asset.Colors.Text.normal.asColor() :
+            Asset.Colors.Text.dark.asColor()
+    }
+    
+    var contractInfo: JSON? {
+        if let redPacketConstantURL = Bundle.main.url(forResource: "red-packet", withExtension: "json"),
+           let data = try? Data(contentsOf: redPacketConstantURL) {
+            return try? JSON(data: data)
+        } else {
+            return nil
+        }
+    }
+    
+    var luckyDropAddressStr: String? {
+        let chainKey = maskUserDefaults.network.redPacketConstantKey
+        return contractInfo?["HAPPY_RED_PACKET_ADDRESS_V4"][chainKey].string
+    }
+    
+    var confirmTitle: String {
+        buttonType.title(token: token, mode: mode, amount: "\(totalQuantityStr) \(token?.symbol ?? "")")
+    }
+    
+    var confirmEnable: Bool {
+        buttonType.isEnable
+    }
+    
+    var tokenURL: URL? {
+        URL(string: token?.logoUrl ?? "")
+    }
+    
+    var buttonAnimating: Bool {
+        buttonType.animating
+    }
+    
+    var showQuantityError: Bool {
+        let quantity = NSDecimalNumber(string: quantityStr)
+        guard quantity != .notANumber else {
+            return false
+        }
+        return quantity.doubleValue > 255
+    }
+    
+    var amountPlaceholder: String {
+        if mode == .average {
+            return L10n.Plugins.Luckydrop.amountPerShare
+        } else {
+            return L10n.Plugins.Luckydrop.totalAmount
+        }
+    }
+    
+    // MARK: - Private property
+    private var disposeBag = Set<AnyCancellable>()
+    @InjectedProvider(\.walletAssetManager)
+    private var walletAssetManager: WalletAssetManager
+    @InjectedProvider(\.userDefaultSettings)
+    private var settings
+    @InjectedProvider(\.vault)
+    private var vault
+    @InjectedProvider(\.mainCoordinator)
+    private var mainCoordinator
+    @InjectedProvider(\.personaManager)
+    private var personaManager
+    private var checkApprovePromise: Promise<Void>?
+    
+    private var nativeTokenAddress: JSON? {
```

Comment:
Use [thsi](https://github.com/DimensionDev/Mask-iOS/pull/63/files#diff-433d9d593c92ea852152de26305b3ff4cf6ed40a53cf472b9772d578632cdb26R152) instead, no need to load this token.json.